rmw_iceoryx_cpp/src/internal/iceoryx_topic_names_and_types.cpp

```diff
@@ -140,6 +143,46 @@ std::map<std::string, std::string> get_topic_names_and_types()
   return names_n_types;
 }
 
+std::map<std::string, std::string> get_service_names_and_types()
+{
+  std::map<std::string, std::string> names_n_types;
+  // Currently, no possibility to query for clients..
+  std::map<std::string, std::vector<std::string>> clients_topics;
+  // Can be discovered via 'ServiceDiscovery'
+  std::map<std::string, std::vector<std::string>> servers_topics;
+  // Currently, no possibility to query for clients..
+  std::map<std::string, std::vector<std::string>> topic_clients;
+  // Can be discovered, but missing associated node
+  std::map<std::string, std::vector<std::string>> topic_servers;
+
+  std::vector<iox::capro::ServiceDescription> available_servers;
+
+  iox::runtime::ServiceDiscovery service_discovery;
+  service_discovery.findService(
+    iox::cxx::nullopt,
+    iox::cxx::nullopt,
+    iox::cxx::nullopt,
+    [&](auto & service_description) {available_servers.push_back(service_description);},
+    iox::popo::MessagingPattern::REQ_RES);
```

Comment:
@elBoberido That is a trade-off in terms of wasting memory vs. reallocation time. If the capacity is not large enough `std::vector` uses the strategy to allocate twice the old capacity on `push_back`, hence this should be a sufficient trade-off here.

Reply:
I was also thinking about the time/memory tradeoff but after estimating the memory I had a tendency towards my proposal. I doesn't matter that much anyway since this should not be in a performance critical path :)